modules/bitcoind.nix

```diff
@@ -239,7 +251,7 @@ in {
         // (if cfg.enforceTor
           then nix-bitcoin-services.allowTor
           else nix-bitcoin-services.allowAnyIP
-        );
+        ) // optionals config.services.lnd.enable nix-bitcoin-services.allowAnyProtocol;  # FOR ZMQ
```

Comment:
Finally figured out that this line is at fault (`--show-trace` wasn't helpful at all). `optionals` is for lists, so if lnd is not enabled it returns an empty list. I think we can use `optionalAttrs` (https://github.com/NixOS/nixpkgs/blob/master/lib/attrsets.nix#L322).

Reply:
ty for testing and investigating this! 🙏🏻

I’ll take a another look at the whole PR later today.